### ISSUE TITLE
`Shift + Alt + Q` does not quit to project if you are editing a text field

### ISSUE THREAD

**Issue by tavurth** [godotengine/godot#58467]
### Godot version

v3.4.3.rc1.official [894b6d504]

### System information

Mac OSX (Intel 2019)

### Issue description

When writing a module or tool script it's quite common to have to restart the editor.

I usually use `Shift + Alt + Q` to perform this and the flow is quite fast, however if you are editing any kind of text field then the latin character `Œ` will be printed instead.

Since I don't think anyone will ever want to actually type this character into a text box, it makes a bit more sense to always quit to project list when we see this keybinding.

### Steps to reproduce

1. Open script editor
2. Press `Alt + Shift + Q`

1. Click on any text box (`LIneEdit`)
2. Press `Alt + Shift + Q`

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
I'm not sure if this behavior can (or should) be overridden for specific key combinations. It would be very frustrating if you rebound some shortcuts around and some of them would never be enterable in a text field.

**Comment by tavurth** [comment]
@Calinou seems similar to this issue https://github.com/godotengine/godot/issues/56961 where it was marked as a usability bug, this seems to fit the same pattern.

**Comment by Calinou** [comment]
> Also this was marked as platform:macos could you check what the windows or linux behaviour of this is?

It mainly depends on the keyboard layout, but also on the OS. Pressing <kbd>Alt + Shift + Q</kbd> on Linux with an AZERTY keyboard layout (fr-oss) does absolutely nothing here.

**Comment by tavurth** [comment]
I see, so it's also not quitting to the project menu?

Actually I don't care about it printing that extra latin character. I mostly care that if I want to restart the editor first I have to switch to the 3d or 2d viewport to restart to the project menu. It's quite a distraction.

**Comment by Calinou** [comment]
> I see, so it's also not quitting to the project menu?

[The Quit To Project List shortcut is bound to <kbd>Ctrl + Shift + Q</kbd> on non-macOS platforms. On macOS, it's overridden to <kbd>Alt + Shift + Q</kbd>, presumably because it conflicts with an OS shortcut.](https://github.com/godotengine/godot/blob/cfc7c78732198a7ebc43b92ed24e7025fdf4bc11/editor/editor_node.cpp#L6472-L6473)
(There is no comment stating *why* the override is necessary, so I'm just guessing.)

**Comment by tavurth** [comment]
Oh ok that's strange. I just tried `Ctrl + Shift + Q` on my mac and there is no response from the editor (not printing any special characters).

Perhaps this bug would be better named: `Mac OS shortcut is inconsistent with cross platform shortcut`?

However I still believe that it should quit to project list even when I have a text field selected.

**Comment by Calinou** [comment]
> If It's really Ctrl + Shift + Q on windows, then it would make sense to match that on OSX.

No, as shortcuts that use <kbd>Ctrl</kbd> on Windows/Linux should *always* use <kbd>Cmd</kbd> on macOS. Not performing this translation is considered poor usability by macOS users.

**Comment by tavurth** [comment]
> I don't think making an exception for this particular case is logical.

However it seems an exception exists if the editor will correctly quit to project menu from the script editor on Windows?

**Comment by Calinou** [comment]
> However it seems an exception exists if the editor will correctly quit to project menu from the script editor on Windows?

Windows does not have special character input when holding <kbd>Ctrl + Shift</kbd> (just like macOS). This only happens because <kbd>Alt + Shift</kbd> is used on macOS.

**Comment by tavurth** [comment]
Perhaps it makes more sense to change the shortcut systemwide (i.e. Windows & Linux & OSX)?

**Comment by tavurth** [comment]
Confirmed that `Ctrl + Shift + Q` functions correctly on OSX even when a `LineEdit` or the `Script Editor` is selected.

Seems like either we should swap the default keybinding on Mac to something else or just swap it everywhere to keep consistency.
